client/src/pages/DocumentPage.jsx

```diff
@@ -167,7 +178,16 @@ export default function DocumentPage() {
     loadContent();
   }, [documentId]);
 
-  const handleTitleChange = (e) => setDocumentTitle(e.target.value);
+  const handleTitleChange = async (e) => {
+    if (e.key === "Enter") {
+      try {
+        await updateDocumentTitle(documentId, documentTitle);
+        e.target.blur();
+      } catch {
+        console.log("Failed to update document title");
+      }
+    }
+  };
```

Comment:
Not for this, but a suggestion for your future projects: This is a decent type of thing to build a custom component for.  Lets say its called `EnterAwareTextBox` and you would use it like this:

```
<EnterAwareTextBox text="" onEnter={handleTitleChange} />
```

so that you can save the "enter" handling logic to use elsewhere.